src/telemetry/Manager.cc

```diff
@@ -38,6 +53,449 @@ void Manager::InitPostBrokerSetup(broker::endpoint& ep)
+		r->Assign(opts_idx, opts_record);
+
+		auto histogram_data = std::get<DblHistogramData>(histogram);
+		auto values_vec = make_intrusive<zeek::VectorVal>(double_vec_type);
+		double observations = 0;
+
+		for ( const auto& b : histogram_data.buckets )
+			{
+			observations += b.count;
+			values_vec->Append(zeek::make_intrusive<zeek::DoubleVal>(b.count));
+			}
+
+		r->Assign(values_idx, values_vec);
+
+		r->Assign(sum_idx, zeek::make_intrusive<zeek::DoubleVal>(histogram_data.sum));
+		r->Assign(observations_idx, zeek::make_intrusive<zeek::DoubleVal>(observations));
+		}
+	else
+		{
+		opts_record = telemetry_mgr->GetMetricOptsRecord<int64_t>(MetricType::Histogram, family);
+		r->Assign(opts_idx, opts_record);
+
+		auto histogram_data = std::get<IntHistogramData>(histogram);
+		auto values_vec = make_intrusive<zeek::VectorVal>(double_vec_type);
+		auto count_values_vec = make_intrusive<zeek::VectorVal>(count_vec_type);
+		int64_t observations = 0;
+
+		for ( const auto& b : histogram_data.buckets )
+			{
+			observations += b.count;
+			values_vec->Append(zeek::make_intrusive<zeek::DoubleVal>(b.count));
+			count_values_vec->Append(val_mgr->Count(b.count));
+			}
+
+		r->Assign(values_idx, values_vec);
+		r->Assign(count_values_idx, count_values_vec);
+
+		r->Assign(sum_idx, zeek::make_intrusive<zeek::DoubleVal>(histogram_data.sum));
+		r->Assign(observations_idx, zeek::make_intrusive<zeek::DoubleVal>(observations));
+
+		r->Assign(count_sum_idx, val_mgr->Count(histogram_data.sum));
+		r->Assign(count_observations_idx, val_mgr->Count(observations));
+		}
+
+	// This is a bit annoying. The bounds / upper bounds of the individual
+	// buckets are not accessible through the family hdl. We set the
+	// opts$bounds field based on bounds collected from an individual
+	// metric instances.
+	//
+	// As the opts_record is cached by the manager, this should only
+	// happen once.
+	//
+	// This is probalby working around something that should be fixed
+	// in the broker::telemetry level.
+	static auto opts_rt = zeek::id::find_type<zeek::RecordType>("Telemetry::MetricOpts");
+	static auto opts_rt_off_bounds = opts_rt->FieldOffset("bounds");
+	static auto opts_rt_off_count_bounds = opts_rt->FieldOffset("count_bounds");
+	static auto bounds_default_val = opts_rt->FieldDefault(opts_rt_off_bounds);
+
+	auto bounds_field_val = opts_record->GetField(opts_rt_off_bounds);
+	if ( ! bounds_field_val || bounds_field_val == bounds_default_val )
+		{
+		std::visit(
+			[&opts_record](const auto& histogram_data)
+			{
+				auto bounds_vec = make_intrusive<zeek::VectorVal>(double_vec_type);
+				for ( const auto& b : histogram_data.buckets )
+					bounds_vec->Append(zeek::make_intrusive<zeek::DoubleVal>(b.upper_bound));
+
+				opts_record->Assign(opts_rt_off_bounds, bounds_vec);
+
+				// If this is an int64_t histogram, also fill count_bounds
+				using T = std::decay_t<decltype(histogram_data)>;
```

Comment:
It's very unusual to have a typedef named `T` since that's generally a template parameter. I think a more speaking name would avoid such confusion:

```suggestion
				using HistogramType = std::decay_t<decltype(histogram_data)>;
```